File: app/src/main/java/id/kuato/woahelper/util/RAM.java
package id.kuato.woahelper.util;

import android.app.ActivityManager;
import android.content.Context;

public class RAM {

    public final long getTotalMemory(final Context context) {
        final ActivityManager actManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        final ActivityManager.MemoryInfo memInfo = new ActivityManager.MemoryInfo();
        assert null != actManager;
        actManager.getMemoryInfo(memInfo);
        return memInfo.totalMem;
    }

    public final String getMemory(final Context context) {
        final String mem;
        mem = new MemoryUtils().bytesToHuman(getTotalMemory(context));
        return mem;
    }
}

File: app/src/main/java/com/woa/helper/util/RAM.kt
package com.woa.helper.util

import android.app.ActivityManager
import android.content.Context

class RAM {
    private fun getTotalMemory(context: Context): Long {
        val actManager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val memInfo = ActivityManager.MemoryInfo()
        checkNotNull(actManager)
        actManager.getMemoryInfo(memInfo)
        return memInfo.totalMem
    }

    fun getMemory(context: Context): String {
        val mem: String = MemoryUtils().bytesToHuman(getTotalMemory(context))
        return mem
    }
}
